addons/arsenal/functions/fnc_onSelChangedLeft.sqf

```diff
@@ -18,429 +19,631 @@ params ["_control", "_curSel"];
 
+// Remove all magazines from the previous weapon that aren't compatible with the new weapon
 private _fnc_clearPreviousWepMags = {
-    private _compatibleMagsBaseWeapon = [_baseWeapon, true] call CBA_fnc_compatibleMagazines;
+    private _compatibleMagsBaseWeapon = compatibleMagazines _baseWeapon;
 
+    // If nothing was selected, remove all magazines from the previous weapon
     if (_item != "") then {
-        _compatibleMagsBaseWeapon = _compatibleMagsBaseWeapon select {!(_x in _compatibleMags)};
+        _compatibleMagsBaseWeapon = _compatibleMagsBaseWeapon - _compatibleMags;
     };
 
     {
         GVAR(center) removeMagazines _x;
-    } foreach _compatibleMagsBaseWeapon;
+    } forEach _compatibleMagsBaseWeapon;
+
+    // Update currentItems
+    private _loadout = getUnitLoadout GVAR(center);
 
-    GVAR(currentItems) set [15, uniformItems GVAR(center)];
-    GVAR(currentItems) set [16, vestItems GVAR(center)];
-    GVAR(currentItems) set [17, backpackItems GVAR(center)];
+    GVAR(currentItems) set [IDX_CURR_UNIFORM_ITEMS, (_loadout select IDX_LOADOUT_UNIFORM) param [1, []]];
+    GVAR(currentItems) set [IDX_CURR_VEST_ITEMS, (_loadout select IDX_LOADOUT_VEST) param [1, []]];
+    GVAR(currentItems) set [IDX_CURR_BACKPACK_ITEMS, (_loadout select IDX_LOADOUT_BACKPACK) param [1, []]];
 };
 
+// Check which right panel has changed
 switch (GVAR(currentLeftPanel)) do {
+    // Primary weapon
+    case IDC_buttonPrimaryWeapon: {
+        private _baseWeapon = (GVAR(currentItems) select IDX_CURR_PRIMARY_WEAPON) call EFUNC(common,baseWeapon);
 
-    case IDC_buttonPrimaryWeapon : {
-        private _baseWeapon = ((GVAR(currentItems) select 0) call bis_fnc_baseWeapon);
-
+        // If nothing selected, remove primary weapon and its magazines
         if (_item == "") then {
             call _fnc_clearPreviousWepMags;
 
             GVAR(center) removeWeapon (primaryWeapon GVAR(center));
-            GVAR(currentItems) set [18, ["", "", "", "", "", ""]];
-            GVAR(currentItems) set [0, _item];
+            GVAR(currentItems) set [IDX_CURR_PRIMARY_WEAPON_ITEMS, ["", "", "", "", "", ""]];
+            GVAR(currentItems) set [IDX_CURR_PRIMARY_WEAPON, ""];
 
             TOGGLE_RIGHT_PANEL_HIDE
         } else {
-            if ((GVAR(currentItems) select 0) != _item && {_baseWeapon != _item}) then {
-                private _compatibleItems = (_item call bis_fnc_compatibleItems) apply {tolower _x};
-                private _cfgMags = configFile >> "CfgMagazines";
-                private _compatibleMags = ([_item, true] call CBA_fnc_compatibleMagazines) select { getNumber (_cfgMags >> _x >> "scope") == 2 };
+            // Check if a new primary weapon was selected
+            if ((GVAR(currentItems) select IDX_CURR_PRIMARY_WEAPON) != _item && {_baseWeapon != _item}) then {
+                // Get magazines that are compatible with the new weapon
+                private _compatibleMags = compatibleMagazines _item;
```

Comment:
Same comment as for previous `compatibleMagazines`.

Reply:
Decision has been taken to move forward with engine commands instead of CBA. If bugs are found with engine commands, ACE will revert back to the CBA functions.